components/kyma-environment-broker/internal/storage/driver/memory/runtime_state.go

```diff
@@ -59,3 +59,12 @@ func (s *runtimeState) GetByOperationID(operationID string) (internal.RuntimeSta
 
 	return internal.RuntimeState{}, dberr.NotFound("runtime state with operation ID %s not found", operationID)
 }
+
+func (s *runtimeState) GetLastByRuntimeID(runtimeID string) (internal.RuntimeState, error) {
+	states, err := s.ListByRuntimeID(runtimeID)
+	if err != nil {
+		return internal.RuntimeState{}, err
+	}
+
+	return states[0], nil
```

Comment:
nit: when integrating this with https://github.com/kyma-project/control-plane/pull/825, I found it useful to guard for slice boundaries and return error self-explanatory error if the `states` slice was empty. I realize that practically this should never occur in real life unless someone truncates `runtime_states` db, but in case something really strange happens and `states` is empty, this code will throw panic when it can instead return `error`.

Reply:
Good point, I will add a db content existence check